sqlglot/dialects/tsql.py

```diff
@@ -365,6 +371,63 @@ class Parser(parser.Parser):
 
         CONCAT_NULL_OUTPUTS_STRING = True
 
+        def _parse_commit_or_rollback(self) -> exp.Commit | exp.Rollback:
+            """Applies to SQL Server and Azure SQL Database
+                COMMIT [ { TRAN | TRANSACTION }
+                    [ transaction_name | @tran_name_variable ] ]
+                    [ WITH ( DELAYED_DURABILITY = { OFF | ON } ) ]
+                    [ ; ]
+
+                ROLLBACK { TRAN | TRANSACTION }
+                    [ transaction_name | @tran_name_variable
+                    | savepoint_name | @savepoint_variable ]
+                    [ ; ]
+
+            Returns:
+                exp.Commit | exp.Rollback: _description_
+            """
+            rollback = self._prev.token_type == TokenType.ROLLBACK
+
+            self._match_texts({"TRAN", "TRANSACTION"})
+            txn_name = self._parse_id_var()
+
+            durability = None
+            if self._match_pair(TokenType.WITH, TokenType.L_PAREN):
+                self._match_text_seq("DELAYED_DURABILITY")
+                self._match(TokenType.EQ)
+                if self._match_text_seq("OFF"):
+                    durability = False
+                else:
+                    self._match(TokenType.ON)
+                    durability = True
+
+                self._match_r_paren()
+
+            if rollback:
+                return self.expression(exp.Rollback, this=txn_name)
+
+            return self.expression(exp.Commit, this=txn_name, durability=durability)
+
+        def _parse_transaction(self) -> exp.Transaction:
+            """Syntax:
+                BEGIN { TRAN | TRANSACTION }
+                [ { transaction_name | @tran_name_variable }
+                [ WITH MARK [ 'description' ] ]
+                ]
+                [ ; ]
+
+            Returns:
+                exp.Transaction: _description_
+            """
+            mark = None
+
+            txn_name = self._parse_id_var()
+
+            if self._match_text_seq("WITH", "MARK"):
+                mark = self._parse_string()
+
+            return self.expression(exp.Transaction, this=txn_name, mark=mark)
```

Comment:
Nit: can also write this a bit more compactly

```python
transaction = self.expression(exp.Transaction, this=self._parse_id_var())
if self._match_text_seq("WITH", "MARK"):
    transaction.set("mark", self._parse_string())

return self.expression(exp.Transaction
```

Reply:
Would your return be:
```python
return transaction
```

Reply:
That's correct, made a typo. Edited.

Reply:
Done, pushed.